In [1]:
import numpy as np
import pandas as pd
import scipy
import statsmodels.api as api
from statsmodels.formula.api import ols
import statsmodels.api as sm
import scipy.stats as stats
from pandas import DataFrame,Series

In [90]:
os=pd.Series([57,103,59,75,84,73,35,110,44,82,67,64,78,53,41,39,80,87,73,65,28,62,49,84,63,77,67,101,91,50])
ns=pd.Series([62,122,54,82,84,86,32,104,38,107,84,85,99,39,34,58,73,53,66,78,41,71,38,95,81,58,75,94,100,68])

len(os)


30

In [91]:
len(ns)

30

In [92]:
os=os*1000

In [93]:
ns=ns*1000

In [94]:
sp=np.arange(1,31)

In [180]:
data = pd.DataFrame({'Sales Person':sp,
                     'Old':os.values,
                     'New':ns.values})
data

Sales Person     Old     New
0              1   57000   62000
1              2  103000  122000
2              3   59000   54000
3              4   75000   82000
4              5   84000   84000
5              6   73000   86000
6              7   35000   32000
7              8  110000  104000
8              9   44000   38000
9             10   82000  107000
10            11   67000   84000
11            12   64000   85000
12            13   78000   99000
13            14   53000   39000
14            15   41000   34000
15            16   39000   58000
16            17   80000   73000
17            18   87000   53000
18            19   73000   66000
19            20   65000   78000
20            21   28000   41000
21            22   62000   71000
22            23   49000   38000
23            24   84000   95000
24            25   63000   81000
25            26   77000   58000
26            27   67000   75000
27            28  101000   94000
28            29   91000  100000
29            30   50000   68000

In [181]:
round(np.mean(data.Old),2)

68033.33

In [182]:
round(np.mean(data.New),2)

72033.33

# 1.Find the mean of old scheme and new scheme column. 

#### Mean of Old Scheme is 68033.33


#### Mean of New Scheme is 72033.33

# 2. Use the five percent significance test over the data to determine the p value to check new scheme has significantly raised outputs? 

# 
Alpha = 5% = 0.05
Ho =  Mean of Old Scheme - Mean of New Scheme = 0 
H1 =  Mean of Old Scheme - Mean of New Scheme < 0 

In [98]:
from    scipy.stats             import  ttest_1samp,ttest_ind, wilcoxon, ttest_ind_from_stats
import  scipy.stats             as      stats  
from    statsmodels.stats.power import  ttest_power
import  matplotlib.pyplot       as      plt

In [208]:
D  = data.New-data.Old

In [209]:
D

0      5000
1     19000
2     -5000
3      7000
4         0
5     13000
6     -3000
7     -6000
8     -6000
9     25000
10    17000
11    21000
12    21000
13   -14000
14    -7000
15    19000
16    -7000
17   -34000
18    -7000
19    13000
20    13000
21     9000
22   -11000
23    11000
24    18000
25   -19000
26     8000
27    -7000
28     9000
29    18000
dtype: int64

In [210]:
t_statistic, p_value  =  stats.ttest_1samp(D,0)
print('P Value %1.3f' % p_value)    

P Value 0.131



#### In this example, p value is 0.131 and it is more than 5% level of significance
#### So we fail to Reject Null Hypothesis at 5% level of significance.

### So there is no sufficient evidence to reject the null hypothesis that there is no significant diference in sales in old Scheme and sales in new scheme i.e there is no significant rise in the output after implementing the scheme

# 3. What conclusion does the test (p-value) lead to? 

## The Probability that Ho is True is 13.1 % i.e Probability that there is no significant difference in sales in old scheme and new scheme is 13.1%,  there is enough support for Ho

The Company has not improved their sales after implementing the new scheme, new scheme is a failure

# 4. Suppose it has been calculated that in order for Titan to break even, the average output must increase by £5000 in the scheme compared to the old scheme. If this figure is alternative hypothesis, what is:
        a) The probability of a type 1 error? 
        b) What is the p- value of the hypothesis test if we test for a difference of $5000? 

# a) P(Reject null hypothesis |  H0  is true) =  α


##### Ho =  Mean of Old Scheme - Mean of New Scheme = 0 
##### H1 =  Mean of Old Scheme - Mean of New Scheme < -5000 

In [110]:
newmean=np.mean(data.New)+5000
newmean

77033.33333333333

In [102]:
ndf=data

In [168]:
nsp=np.arange(1,31)
nns=pd.Series([62,122,54,82,84,86,32,104,38,107,84,85,99,39,34,58,73,53,66,78,41,71,38,95,81,58,75,94,100,68])
nos=pd.Series([57,103,59,75,84,73,35,110,44,82,67,64,78,53,41,39,80,87,73,65,28,62,49,84,63,77,67,101,91,50])
nns=(nns*1000)+5000
nos=nos*1000

In [166]:
nns.mean()

77033.33333333333

In [169]:
ndata = pd.DataFrame({'Sales Person':nsp,
                     'Old':nos.values,
                     'New':nns.values})
ndata

Sales Person     Old     New
0              1   57000   67000
1              2  103000  127000
2              3   59000   59000
3              4   75000   87000
4              5   84000   89000
5              6   73000   91000
6              7   35000   37000
7              8  110000  109000
8              9   44000   43000
9             10   82000  112000
10            11   67000   89000
11            12   64000   90000
12            13   78000  104000
13            14   53000   44000
14            15   41000   39000
15            16   39000   63000
16            17   80000   78000
17            18   87000   58000
18            19   73000   71000
19            20   65000   83000
20            21   28000   46000
21            22   62000   76000
22            23   49000   43000
23            24   84000  100000
24            25   63000   86000
25            26   77000   63000
26            27   67000   80000
27            28  101000   99000
28            29   91000  105000
29            30   50000   73000

In [211]:
D1  = ndata.New- ndata.Old

In [212]:
D1

0     10000
1     24000
2         0
3     12000
4      5000
5     18000
6      2000
7     -1000
8     -1000
9     30000
10    22000
11    26000
12    26000
13    -9000
14    -2000
15    24000
16    -2000
17   -29000
18    -2000
19    18000
20    18000
21    14000
22    -6000
23    16000
24    23000
25   -14000
26    13000
27    -2000
28    14000
29    23000
dtype: int64

In [221]:
t_statistic, p_value  =  stats.ttest_1samp(D-5000,0)
print('P Value %1.3f' % p_value) 

P Value 0.700


In [216]:
p=1-stats.t.cdf(t_statistic,30)
p

0.6499803739748564

##### if we increase the average return of the new scheme by 5000, we can see that there is a decrease in the p value, Now P vaule is less than Alpha(0.05),therefore we reject the Ho. So there is a significant change in the sales after implementing the new scheme and breaking even.

#### P(Reject null hypothesis | H0 is true) = 0.002

#### we control the Type I error by determining the risk level, alpha, the level of significance that we are willing to reject 

# Probability of Type1 Error is 0.2%

# b) What is the p- value of the hypothesis test if we test for a difference of $5000? 

P Value 0.002 i.e 0.2 %
